File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters();
int count_words();
int count_sentences();


int main(void)
{

    //get Text
    string text = get_string("Text: ");
    //count letters, words and sentences
    float letters = (float)count_letters(text);
    float words = (float)count_words(text);
    float sentences = (float)count_sentences(text);
    //calculate index
    float l = letters / words * 100;
    float s = sentences / words * 100;
    float index = 0.0588 * l - 0.296 * s - 15.8;

    //round index to integer
    int grade = rint(index);

    //print grade
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}

//count letters
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

//count words
int count_words(string text)
{
    int words = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words + 1;
}

//count sentences
int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    l = 100 * letters / words
    coleman_Liau_index = 0.0588 * l - 0.296 * sentences - 15.8



    print(f"Letters: ", letters)
    print(f"Words: ", words)
    print(f"Sentences: ", sentences)

    print (coleman_Liau_index)

def count_letters(text):
    letters = 0
    for c in text:
        if "A" <= c <= "Z" or "a" <= c <= "z":
            letters += 1
    return letters

def count_words(text):
    words = 0
    for c in text:
        if c == " ":
            words += 1
    return words +1

def count_sentences(text):
    sentences = 0

    #period, exclamation point, or question mark indicates the end of a sentence
    end = [".", "!", "?"]
    for c in text:
        if c in end:
            sentences += 1
    return sentences

main()

# 400 buchstaben / 200 wörter